flang/lib/Optimizer/Transforms/OMPMapInfoFinalization.cpp

```diff
@@ -219,6 +430,33 @@ class OMPMapInfoFinalizationPass
     mapClauseOwner.getMapOperandsMutable().assign(newMapOps);
```

Comment:
Not part of the PR (sorry for the extra comments), but I have a feeling something is not correct here. Doesn't this line overwrite all map operands which means if we have map clauses from multiple structures/types we will end up with the map ops from the last one and its members, right?

Shouldn't we instead append to a shared vector and assign the whole vector after we walk all map info ops (line 5.4)? Apologies if I missed something.